test/test-server/cloud_kafka_handlers.go

```diff
@@ -57,14 +56,18 @@ func handleKafkaClientQuota(t *testing.T) http.HandlerFunc {
 				_, err := io.WriteString(w, "")
 				require.NoError(t, err)
 			case http.MethodPatch:
+				quotaPatch := &kafkaquotasv1.KafkaQuotasV1ClientQuota{ // make a deep copy so changes don't reflect in subsequent tests
```

Comment:
Can we just change the type from a slice of pointers to a regular slice, and do a normal copy?